lib/ui/fading_app_bar.dart

```diff
@@ -249,21 +249,28 @@ class FadingAppBarState extends State<FadingAppBar> {
   Future<void> _download(File file) async {
     final dialog = createProgressDialog(context, "downloading...");
     await dialog.show();
-    if (file.fileType == FileType.image) {
-      file.localID = (await PhotoManager.editor.saveImageWithPath(
+    if (file.fileType == FileType.image ||
+        file.fileType == FileType.livePhoto) {
+      final savedAsset = (await PhotoManager.editor.saveImageWithPath(
         (await getFile(file)).path,
         title: file.title,
-      )).id;
-    } else {
-      file.localID = (await PhotoManager.editor.saveVideo(
-        (await getFile(file)),
+      ));
+      file.localID = savedAsset.id;
+    } else if (file.fileType == FileType.video) {
+      final savedAsset = (await PhotoManager.editor.saveImageWithPath(
+        (await getFile(file)).path,
         title: file.title,
-      )).id;
+      ));
+      file.localID = savedAsset.id;
     }
     await FilesDB.instance.insert(file);
     await LocalSyncService.instance.trackDownloadedFile(file);
     Bus.instance.fire(LocalPhotosUpdatedEvent([file]));
     await dialog.hide();
-    showToast("file saved to gallery");
+    if (file.fileType != FileType.livePhoto) {
+      showToast("file saved to gallery");
+    } else {
+      showToast("only image saved, live photo support coming soon");
```

Comment:
If it's not too much effort, I feel we can give the user an option to choose between the photo and the video. Not sure when we'll come back to add this.